File: src/main/java/leetcode/arrays/ContainerWithMostWater.java
package main.java.leetcode.arrays;

public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea =0;

        while(left < right){
            int w = right - left;
            int h = Math.min(height[left], height[right]);
            int area = w * h;

            maxArea = Math.max(area, maxArea);

            if(height[right] > height[left]){
                left++;
            }else{
                right --;
            }
        }

        return maxArea;
    }
}

File: src/main/java/nvidia/ContainerWithMostWater.cpp
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int maxArea =0;

        while(left< right){
           int w = right - left;
            int h = min(height[left], height[right]);
            int currentArea = w * h;
            
            maxArea = max(maxArea, currentArea);

            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxArea;
    }
};